BUG #1944: Rename line to linetype for isoline, unified1d and vector.

danlipsa opened the pull request:
Add separate function to set type, color and width from a line.

danlipsa commented:
@doutriaux1 @aashish24 Please review. This required more changes than I thought or wanted. Overall I think it is a much clearer API.

sankhesh commented:
Do: test

kwrobot01 commented:
*Warning*: Requests to test ignored; user lacks testing privileges:  
@sankhesh: test   

**SERVER ERROR**: project uvcdat directory not available! @sankhesh

kwrobot01 commented:
*Warning*: Requests to test ignored; user lacks testing privileges:  
@sankhesh: test   

**SERVER ERROR**: project uvcdat directory not available! @sankhesh

kwrobot01 commented:
**SERVER ERROR**: project uvcdat directory not available! @sankhesh

kwrobot01 commented:
**SERVER ERROR**: project uvcdat directory not available! @sankhesh

sankhesh commented:
Do: check

danlipsa commented:
Do: check

danlipsa commented:
@doutriaux1 @aashish24 ping! This also has been setting here for too long.

danlipsa commented:
Do: test

danlipsa commented:
@sankhesh @doutriaux1 This test
https://open.cdash.org/testDetails.php?test=462274077&build=4475133
uses
ACTIVATING ENV:2.6
but should be something like:
ACTIVATING ENV:2.6-11-g94c840d
Any idea why?

doutriaux1 commented:
That is odd indeed did you start with the 2.6 env ?

danlipsa commented:
This is done by buildbot. Sankhesh says that he deletes both the source and the build directory so I am not sure where 2.6 is picked up from.

danlipsa commented:
@doutriaux1 This is done by buildbot. Sankhesh says that he deletes both the source and the build directory so I am not sure where 2.6 is picked up from.

danlipsa commented:
Do: test

danlipsa commented:
Do: test

aashish24 commented on testing/vcs/test_vcs_set_line_attribute_from_line.py:
```diff
@@ -6,16 +6,22 @@
 
 v=vcs.createvector()
-v.line = "vcs_test_set_line"
+v.setLineAttributes("vcs_test_set_line")
```
@danlipsa what does this function do at this point in the code?

danlipsa replied:
Sets linetype, linecolor and linewidth from the line with the name 'vcs_test_set_line'.

aashish24 commented on testing/vcs/test_vcs_set_line_attribute_from_line.py:
```diff
@@ -6,16 +6,22 @@
 assert(v.linecolor == 242)
 assert(v.linewidth == 5.6)
-assert(v.line == "dash")
+assert(v.linetype == "dash")
```
+1

aashish24 commented on Packages/vcs/vcs/vector.py:
```diff
@@ -507,14 +522,17 @@ def _setlinecolor(self, value):
+        self._linetype = value
+    linetype = property(_getlinetype, _setlinetype)
+
+    def setLineAttributes(self, line):
```
@danlipsa should we add docstring?

danlipsa replied:
@aashish24 I missed that. If you search for 'def setLineAttributes' you'll find the docstring. I'll have to copy it in a few more places.

doutriaux1 replied:
@danlipsa @aashish24 speaking of doc strings I asked our interns here to atually add them to the propertied objects as well. I will be a nice addition

danlipsa commented:
@aashish24 Done.

doutriaux1 commented on Packages/vcs/vcs/isoline.py:
```diff
@@ -619,28 +619,60 @@ def _setlinecolors(self, value):
-        except:
-            pass
-
-    line = property(_getline, _setline)
```
I don't think we should break backward compatibility, we should probably keep it and raise a deprecated warning.

danlipsa replied:
Makes sense. I will do that.

danlipsa replied:
@doutriaux1 @doutriaux1 Makes sense. The line interface will still have the bug @chaosphere2112 reported. We won't do any fixes there. Maybe remove it in the next version.

doutriaux1 commented on Packages/vcs/vcs/unified1D.py:
```diff
@@ -46,7 +46,6 @@ def process_src(nm, code, typ):
                 "ymtics#1", "ymtics#2",
                 "xaxisconvert", "yaxisconvert",
                 "datawc_tunits",
-                "Tl",
```
you can't remove this, `scr` files are old files if the attribute is here that means it probably was part of the scr files.

danlipsa replied:
@doutriaux1 Note that I deal with "Tl" later on, look at the full source code. That way I use the new interface not the deprecated one.

doutriaux1 commented on Packages/vcs/vcs/unified1D.py:
```diff
@@ -496,16 +510,21 @@ def _setlinecolor(self, value):
-                value = value.name
-            value = VCS_validation_functions.checkLineType(self, 'line', value)
-        self._line = value
-    line = property(_getline, _setline)
```
same here i think we should preserve backward compatibility for now and issue a deprecated warning

danlipsa replied:
@doutriaux1 Sounds good.

doutriaux1 commented on Packages/vcs/vcs/utils.py:
```diff
@@ -312,42 +312,6 @@ def _scriptrun(script, canvas=None):
                 vcs.process_src_element(s)
                 processing = False
     f.close()
-    # Ok now we need to double check the isolines
```
@danlipsa why is this removed?

danlipsa replied:
@doutriaux1 I did not really understand that code, and something bad was happening there as far as I can remember. It made all tests pass. I will take a closer look.

doutriaux1 replied:
@danlipsa without taking a deep look I can tell you what I "remember" 
isoline graphic methods that had styles and colors used to be stored in the old `.scr` files as separte `line` objects.
I think this bit of code as something to do with this.

doutriaux1 commented on Packages/vcs/vcs/vector.py:
```diff
@@ -59,9 +59,7 @@ def process_src(nm, code):
         sp = scode.split("=")
         nm = sp[0].strip()
         nm = nm.replace("#", "")
-        if nm == "Tl":
```
same here it will break reading in old `scr` files I think.

danlipsa replied:
@doutriaux1 Same as before, I deal with 'Tl' later in the code using the new interface.

doutriaux1 commented on Packages/vcs/vcs/vector.py:
```diff
@@ -190,11 +205,11 @@ class Gv(object):
-     vc.line=2              # Same as vc.line='dot'
-     vc.line=3              # Same as vc.line='dash-dot'
-     vc.line=4              # Same as vc.line='long-dot'
+     vc.linetype=0              # Same as vc.linetype='solid'
```
should we simply set them to 'solid', etc.. it's easier to read no?

danlipsa replied:
@doutriaux1 This is a comment in the sources, the user should use strings instead of numbers indeed.

doutriaux1 commented on testing/vcs/test_vcs_dump_json.json:
```diff
@@ -220,8 +224,8 @@
    "datawc_y2": 1e+20, 
-   "line": null, 
    "linecolor": null, 
+   "linetype": null,
```
oh yes that is why we need to keep backward compatibilty for a while, all the json file have `line` in them. That will break them all.

doutriaux1 commented on testing/vcs/test_vcs_set_line_attribute_from_line.py:
```diff
@@ -6,16 +6,22 @@
+assert(yx.linetype == "dash")
+
+iso = vcs.createisoline()
+iso.setLineAttributes([l, "solid", l])
```
maybe "solid" isn't the best choice here, it could llead people to think they can pass a line `type` instead of a line name. I would use something very different like: `thick` or `pink`

danlipsa replied:
I'll just add a comment.

danlipsa commented on Packages/vcs/vcs/utils.py:
```diff
@@ -312,42 +312,6 @@ def _scriptrun(script, canvas=None):
-            continue
-        for att in ["line", "textcolors", "text"]:
-            try:
-                setattr(g, att, getattr(g, att))
```
@doutriaux1 What does this code do?

danlipsa commented on Packages/vcs/vcs/utils.py:
```diff
@@ -312,42 +312,6 @@ def _scriptrun(script, canvas=None):
-                            lst.append(e)
-                elif att == "text":
-                    for e in g.line:
-                        if e in vcs.elements["textorientation"]:
```
@doutriaux1 Why is e from g.line is search in vcs.elements['textorientation']?

doutriaux1 replied:
because textorientation objects contain the relevant information to this attribute so it is possible that the use rsent us a list of textorientation to look for. We could probably drop this and the `texttable` bellow and simply look in `textcombined` it would be easier to understand and maintain.

danlipsa commented on Packages/vcs/vcs/utils.py:
```diff
@@ -312,42 +312,6 @@ def _scriptrun(script, canvas=None):
-                            lst.append(e)
-                elif att == "textcolors":
-                    for e in g.line:
-                        if e in vcs.elements["texttable"]:
```
@doutriaux1 Same question here: what is the link between e from g.line (which is a linestyle) and vcs.elements['texttable']?

doutriaux1 replied:
i think you're right it should be `g.text` and `g.textcolor` looks like some copy/pasting that went bad.

danlipsa commented:
@doutriaux1 @aashish24 I still don't understand the code I removed from utils.py - it does not affect the tests. Maybe you guys can help me here. I added back the line interface which prints a deprecated message and calls on the new linetype/setLineAttributes interface. Please review

danlipsa commented:
These are the failures on linux:
29 tests failed out of 611:

	 13 - testWrite (Failed)
	 62 - test_vcs_read_old_scr_2 (Failed)
	 73 - test_vcs_verify_boxfill_basics (Failed)
	121 - test_vcs_boxfill_decreasing_latitude (Failed)
	122 - test_vcs_meshfill_no_wrapping (Failed)
	413 - test_vcs_isofill_level0 (Failed)
	414 - test_vcs_isofill_level1 (Failed)
	415 - test_vcs_isofill_level2 (Failed)
	469 - test_vcs_matplotlib_colormap (Failed)
	549 - dv3d_vector_test (Failed)
	550 - dv3d_slider_test (Failed)
	551 - Hovmoller_volume_test (Failed)
	552 - dv3d_volume_test (Failed)
	553 - dv3d_constituents_test (Failed)
	554 - dv3d_surface_test (Failed)
	555 - dv3d_hovmoller_test (Failed)
	580 - cdms2_test_createcopy_lose_dtype (Failed)
	581 - CDMS_Test_01 (Failed)
	582 - CDMS_Test_02 (Failed)
	583 - CDMS_Test_03 (Failed)
	584 - CDMS_Test_04 (Failed)
	590 - CDMS_Test_10 (Failed)
	598 - CDMS_Test_18 (Failed)
	599 - CDMS_Test_20 (Failed)
	601 - CDMS_Test_multiple_formats (OTHER_FAULT)
	602 - CDMS_Test_write_compressed (Failed)
	606 - CDMS_Test_pack (Failed)
	609 - CDMS_Test_write_extended_dimension (Failed)
	610 - CDMS_Test_del_attributes (Failed)

doutriaux1 commented:
@danlipsa I will look into this, but it maybe easier to look at this together when you're here next week, I'm not sure why you removed it at all.

danlipsa commented:
Do: test

danlipsa commented:
@doutriaux1 @aashish24 This is ready! I added back the removed code and the line interface.

danlipsa commented:
This are the test failures on Linux:
96% tests passed, 27 tests failed out of 611

Total Test time (real) = 407.44 sec

The following tests FAILED:
	 13 - testWrite (Failed)
	 73 - test_vcs_verify_boxfill_basics (Failed)
	121 - test_vcs_boxfill_decreasing_latitude (Failed)
	122 - test_vcs_meshfill_no_wrapping (Failed)
	413 - test_vcs_isofill_level0 (Failed)
	414 - test_vcs_isofill_level1 (Failed)
	415 - test_vcs_isofill_level2 (Failed)
	469 - test_vcs_matplotlib_colormap (Failed)
	549 - dv3d_vector_test (Failed)
	550 - dv3d_slider_test (Failed)
	551 - Hovmoller_volume_test (Failed)
	552 - dv3d_volume_test (Failed)
	553 - dv3d_constituents_test (Failed)
	554 - dv3d_surface_test (Failed)
	555 - dv3d_hovmoller_test (Failed)
	580 - cdms2_test_createcopy_lose_dtype (Failed)
	581 - CDMS_Test_01 (Failed)
	582 - CDMS_Test_02 (Failed)
	583 - CDMS_Test_03 (Failed)
	584 - CDMS_Test_04 (Failed)
	590 - CDMS_Test_10 (Failed)
	598 - CDMS_Test_18 (Failed)
	599 - CDMS_Test_20 (Failed)
	602 - CDMS_Test_write_compressed (Failed)
	606 - CDMS_Test_pack (Failed)
	609 - CDMS_Test_write_extended_dimension (Failed)
	610 - CDMS_Test_del_attributes (Failed)

doutriaux1 commented:
@danlipsa no worries about the failed test, it's due to hdf5 not matching anymore by default under conda. But once I merge the legend PR you will likely need to re-update your baselines.

danlipsa commented:
@doutriaux1 Indeed the failed vcs tests are because of hdf5 problem.

danlipsa commented:
@doutriaux1 @aashish24 Lets get this merged as well, unless you have additional comments/concerns.

doutriaux1 commented:
@danlipsa I would like to finish/merge fix_travis first if it doesn't take too long.

danlipsa commented:
@doutriaux1 Do you worry about conflicts/broken tests?

doutriaux1 commented:
kind of the opposite, since it shouldn't break anything I would like to see a PR pass on travis ;)

danlipsa commented:
@doutriaux1 I guess my point was that even if you merge this your PR on travis can go on as before. You should not have any conflicts or any new failing tests. But I can wait if you want to take a closer look at this and you want to finish the travis work first without interruption.

danlipsa commented:
Do: check

danlipsa commented:
Do: test

danlipsa commented:
Do: test

sankhesh commented:
Do: test

aashish24 commented:
@sankhesh I see 4 contents check but no build / test check after the last test command from @danlipsa. What's wrong?

sankhesh commented:
@aashish24 I was puzzled by it yesterday too. I looked into it. This pull request is older than the changes we made to the bot and I am assuming buildbot is getting confused with the different commands on it. I'll talk to Ben on Monday to see what's wrong exactly.

sankhesh commented:
Do: test

sankhesh commented:
Do: test

danlipsa commented:
@sankhesh It seems that some of the tests still do not activate the right environment:
https://open.cdash.org/testDetails.php?test=462274077&build=4495107

sankhesh commented:
@danlipsa All the tests on default build are activating the right environment. The tests on the lean build just activate 2.6. I think that is a UV-CDAT issue that needs to be fixed in the code.

danlipsa commented:
@sankhesh How do you see the default build? I only see the lean build - I click on cdash

sankhesh commented:
Here is the default build: https://open.cdash.org/buildSummary.php?buildid=4495026

danlipsa commented:
@sankhesh The test failures look like @doutriaux1 colorbar baselines. I will run the test on my machine to see if they pass.

danlipsa commented:
@sankhesh On my machine most tests pass - I have 26 failures - most vcs once are from hdf5.
Here are @doutriaux1 baselines.
https://github.com/UV-CDAT/uvcdat-testdata/pull/149/files

In the link you provided, it looks like @doutriaux1 changes are not in, possible because the code us the wrong conda env? The SHA is correct. The baseline is right but the generated image is wrong. Can you look in the LastTest.log to see the conda env for the tests?

doutriaux1 commented:
Do: test
